FastApi/app/routers/post.py

```diff
@@ -170,30 +188,49 @@ def update_post(
 @router.patch("/{id}", response_model=schemas.PostResponse)
 def update_post_patch(
     id: int,
-    post_patch: schemas.PostCreate,
+    file: UploadFile = File(None),
+    post_update: schemas.PatchBase = Depends(),
     db: Session = Depends(get_db),
     current_user: int = Depends(oauth2.get_current_user),
 ):
-
+    
     post_query = db.query(models.Post).filter(models.Post.id == id)
-    patch_post = post_query.first()
+    returned_post = post_query.first()
+
+
+    url=None
+    #check if file exist and format of it
+    if file:
+            if utils.is_file_type_valid(file) == True:
+                url=utils.file_write(file,image_folder_path)
+            else:
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST, detail="Unsupported media!")
+
 
     # making sure that error dont occure if there is no required ID
-    if patch_post == None:
+    if returned_post == None:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
-            detail=f"post with that id {id} does not exist",
-        )
+            detail=f"post with that id {id} does not exist",)
 
-    if patch_post.owner_id != oauth2.get_current_user.id:
+    if returned_post.owner_id != current_user.id:
         raise HTTPException(
             status_code=status.HTTP_403_FORBIDDEN,
-            detail="Not authorized to perform requested action",
-        )
+            detail="Not authorized to perform requested action",)
+    
+
+    if post_update.content != None: 
+        returned_post.content = post_update.content  
+
+    if post_update.title != None: 
+        returned_post.title = post_update.title
 
-    post_query.update(post_patch.model_dump(), synchronize_session=False)
+    if url != None: 
+       returned_post.image_url=url
+  
 
     db.commit()
-    db.refresh(patch_post)
+    db.refresh(returned_post)
```

Comment:
I would suggest to name this property in other way, for example: just "post", since name of request body property is "post_update"